### PR TITLE
Remove constraint on unused llrb package

### PR DESCRIPTION
**Pull request by fejta**
/assign @BenTheElder @ixdy @cblecker @stevekuznetsov 

```
vim Gopkg.lock && hack/update-deps.sh
```

Only used by https://github.com/kubernetes/test-infra/blob/master/vendor/github.com/google/btree/btree_mem.go#L15, which does not actually build

### CODE REVIEW

**Comment by ixdy** [comment]
Shouldn't this also remove the `vendor/github.com/petar/GoLLRB/llrb` directory, then?

**Comment by fejta** [comment]
It does? 

https://github.com/fejta/test-infra/tree/15b761a0e3ce34760b1e9a2a69682e4ef940a173/vendor/github.com

**Comment by ixdy** [comment]
oh, I was looking at my local tree, which was in a weird state, apparently.

/lgtm

File: Gopkg.lock [hunk 1]
```diff
@@ -166,12 +166,6 @@
   packages = ["pbutil"]
   revision = "fc2b8d3a73c4867e51861bbdd5ae3c1f0869dd6a"
 
-[[projects]]
-  branch = "master"
-  name = "github.com/petar/GoLLRB"
```
Comment (ixdy) [inline Gopkg.lock]: I'm a little worried that `dep ensure` will just add this back, though.

Comment (fejta) [reply Gopkg.lock]: I don't think that is how `dep ensure` works

This is a constraint, telling `dep ensure` that regardless of whether or not anything in the repo actually depends on this, check it out

By removing this, we're no longer forcing dep to check it out, so it deletes it for us

Comment (fejta) [reply Gopkg.lock]: https://github.com/golang/dep#removing-dependencies

Comment (ixdy) [reply Gopkg.lock]: I'm still new to `dep`, but I thought you aren't supposed to edit `Gopkg.lock`, only `Gopkg.toml`.

Comment (stevekuznetsov) [reply Gopkg.lock]: Editing the `toml` was my understanding too

Comment (fejta) [reply Gopkg.lock]: Yeah that seems the be what a closer reading of the docs say... Heh

**Pull request merged**